packages/client/src/schedule-helpers.ts

```diff
@@ -342,6 +354,7 @@ export async function decodeScheduleAction(
   throw new TypeError('Unsupported schedule action');
 }
 
+// TODO(thomas): move to internal package (or at least to the helpers.ts file)
```

Comment:
Please also create `encodeSearchAttributes` and `encodeTypedSearchAttributes` help functions. Move those four helper functions, and anything else that is specific to encoding/decoding SAs, to `common/src/converter/payload-search-attributes.ts`. And make sure to use those helpers everywhere (e.g. workflow-client, schedule-client, workflow/src/internals.ts, etc).

Reply:
Actually, it might even be desirable to have an `encodeSearchAttributes` function, which accepts both untyped and typed SAs. There's some little details in how to properly do that; duplicating that code is undesirable.

Reply:
Done.
Yeah that reduced a lot of the duplicated code, didn't realize how much was being re-written.
Wrote `encodeUnifiedSearchAttributes` as a single encoding function for both forms of search attributes.